### PR TITLE
Allows you to set middleware in configuration

### PR DESCRIPTION
**Pull request by Jono20201**
This PR allows you to set a middleware group via the configuration file as well as commenting improvements to the default config.

Also a workaround for https://github.com/GeneaLabs/laravel-caffeine/issues/47 as I don't have the time to find the root cause at the moment.

### CODE REVIEW

File: src/Providers/LaravelCaffeineService.php [hunk 1]
```diff
@@ -30,7 +29,24 @@ public function provides() : array
...
+        $middleware_config = app(Cache::class)->get('genealabs-laravel-caffeine.middleware');
+
+        if ($middleware_config) {
+            return ['middleware' => $middleware_config];
+        }
+
+        return $this->middlewareGroupExists('web') ? ['middleware' => 'web'] : [];
+    }
+
+    protected function middlewareGroupExists(string $group) : bool
```
Comment (Jono20201) [inline src/Providers/LaravelCaffeineService.php]: I have a dislike for using `private` unless it is REALLY needed. In this case someone _may_ want to extend this class and would then be prevented from using the `middlewareGroupExists` method for no reason, as far as I can see.

**Comment by mikebronner** [comment]
Awesome, thanks @Jono20201 -- I will review this over the next few days. I might have a differing opinion of private vs. protected, but I understand where you are coming from. :)

**Comment by Jono20201** [comment]
Haha, no worries. There's lots of little bits I could change in this code, I always find it hard not to make small tweaks when doing OS work. Let me know if you want any changes.

**Comment by Jono20201** [comment]
@mikebronner Any update on getting this pulled in? Would like to switch back to using the official package rather than using my forked version.

Thanks!

**Comment by mikebronner** [comment]
Sorry for the delay ... holidays were crazy, and I had some changes work-wise. Getting back on track now. Will look at this later today. Thanks again. :)

**Comment by Jono20201** [comment]
@mikebronner Resolved a conflict that arose with L5.4 update

**Comment by mikebronner** [comment]
Closing this as PR #76 fulfills this requirement. Thanks for submitting, and sorry it took so long to get resolved!

**Comment by mikebronner** [comment]
To achieve this set `use-route-middleware` to `true` in config file, then add `caffeinated` to your web middleware group. (The route middleware alias is automatically registered.)

**Pull request closed without merging**